### PR TITLE
Adds new client methods for project, secret, & org invite endpoints.

### PR DESCRIPTION
**Pull request by mikejholly**
This PR adds support for the new projects, secrets, and org endpoints in Earthly Cloud. These will act as the basis for the forthcoming command changes.

### CODE REVIEW

File: cloud/client.go [hunk 1]
```diff
@@ -1111,6 +1165,385 @@ func (c *client) DeleteSatellite(ctx context.Context, name, orgID string) error
...
+			CreatedAt:  secret.CreatedAt.AsTime(),
+			ModifiedAt: secret.ModifiedAt.AsTime(),
+		})
+	}
+
+	return secrets, nil
+}
+
+// SetSecret adds or updates the given path and secret combination.
+func (c *client) SetSecret(ctx context.Context, path string, secret []byte) error {
```
Comment (alexcb) [inline cloud/client.go:1400]: What's the difference between this new function and the existing one we have for setting secrets: https://github.com/earthly/earthly/blob/42839f02736cc1e486628915afef9b1304e78adb/cloud/client.go#L714-L730

File: cloud/client.go [hunk 2]
```diff
@@ -1111,6 +1165,385 @@ func (c *client) DeleteSatellite(ctx context.Context, name, orgID string) error
...
+
+// AddProjectMember adds a new member to the project by email or user ID.
+func (c *client) AddProjectMember(ctx context.Context, orgName, name, idOrEmail, permission string) error {
+	u := fmt.Sprintf("/api/v0/projects/%s/%s", orgName, name)
+
+	req := &secretsapi.AddProjectMemberRequest{
+		Permission: permission,
+	}
+
+	if userID, err := uuid.Parse(idOrEmail); err == nil && userID != uuid.Nil {
```
Comment (alexcb) [inline cloud/client.go:1290]: Is it possible to move this `idOrEmail` logic into a separate function, and then extend all the project functions (e.g. `UpdateProjectMember`, `ListProjectMembers`, etc), to all accept an email or ID?

**Review by alexcb — COMMENTED** [review]
looking good, just a couple questions.

File: cloud/client.go (hunk 1 above)
Comment (mikejholly) [reply cloud/client.go:1400]: So @vladaionescu asked me to keep the old code around and add the new methods/commands in parallel until we're ready to cut over. The new `SetSecret` method is just for the new project-based secret update endpoint. Perhaps some sort of prefix would help clarify, but I don't think the older code will be around for too much longer.

File: cloud/client.go (hunk 2 above)
Comment (mikejholly) [reply cloud/client.go:1290]: I think that's only used for that one endpoint/method. The other ones assume a userID, which is returned from the member list endpoint.

**Review by vladaionescu — COMMENTED** [review]
It's a bit difficult to tell which methods are new and which ones are the ones that will be deprecated. I would split this file up in the following way (or something along these lines):

* `oldclient.go` - keep all the old APIs here
* `org.go` - org-related methods
* `project.go` - project-related methods
* `secret.go` - secret-related methods
* `account.go` - login, account registration and keys/tokens management methods
* `common.go` - anything generic that doesn't fit in any of the above

I think this would make this API more manageable as it's already growing out of proportion for a single file.

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**